fn_cve_search/fn_cve_search/components/function_cve.py

```diff
@@ -0,0 +1,269 @@
+            raise FunctionError("CVE Api Call Failed : {}".format(call_err))
+
+    def _browse_cve_api(self, vendor_name=None):
+        """
+        :param vendor_name: Vendor Name to search for Vulnerabilities from CVE DB
+        :return: if vendor_name=None  then returns JSON with all the vendors, otherwise JSON with all the products
+        associated to a vendor
+        """
+        log = logging.getLogger(__name__)
+
+        if vendor_name:
+            _browse_api = '{}/{}'.format(self.cve_base_url.format('browse'), vendor_name)
+        else:
+            _browse_api = self.cve_base_url.format('browse')
+        return self._make_rest_api_get_call(_browse_api, api_call='browse')
+
+    def _search_cve_api(self, vendor_name=None, product=None):
+        """
+        :param vendor_name:  Name of the Vendor
+        :param product:   Name of the Product
+        :return:JSON with all the vulnerabilities per vendor and a specific product
+        """
+        if vendor_name is not None and product is not None:
+            _search_api = '{}/{}/{}'.format(self.cve_base_url.format('search'), vendor_name, product)
+        elif vendor_name is not None and product is None:
+            _search_api = '{}/{}'.format(self.cve_base_url.format('search'), vendor_name)
+        elif vendor_name is None and product is not None:
+            _search_api = '{}/{}'.format(self.cve_base_url.format('search'), product)
+        return self._make_rest_api_get_call(_search_api, api_call='search')
+
+    def _get_specific_cve_data(self, cve_id=None):
+        """
+        :param cve_id: Specific CVE ID
+        :return:a JSON of a specific CVE ID
+        """
+        if cve_id:
+            _specific_cve_url = '{}/{}'.format(self.cve_base_url.format('cve'), cve_id)
+            return self._make_rest_api_get_call(_specific_cve_url, api_call='cve')
+        else:
+            raise FunctionError("Please Specify the CVE ID to search for Vulnerability")
+
+    def _last_30_cves(self):
+        """
+        :return:a JSON of the last 30 CVEs including CAPEC, CWE and CPE expansions
+        """
+        _last_url = '{}'.format(self.cve_base_url.format('last'))
+        return self._make_rest_api_get_call(_last_url, api_call='last')
+
+    def _cve_db_information(self):
+        """
+        :return:more information about the current databases in use and when it was updated
+        """
+        _db_info_url = '{}'.format(self.cve_base_url.format('dbInfo'))
+        return self._make_rest_api_get_call(_db_info_url, api_call='db')
+
+    @function("function_cve")
+    def _function_cve_function(self, event, *args, **kwargs):
+        """Function: A Function to Search  Common Vulnerability Exposures Data from https://cve.circl.lu Data base."""
+        try:
+            # Get the function parameters:
+            cve_search_data = kwargs.get("cve_search_data")  # text
+            cve_search_criteria = kwargs.get("cve_search_criteria")  # text
```

Comment:
this should be a select field in the resilient function. That way the values are pre-selectable in the workflow that contains it.